internal/scheduler/scheduler.go

```diff
@@ -230,21 +230,22 @@ func (s *Scheduler) runJob(ctx context.Context, r *job.Run) error {
 	go func() {
 		defer rj.cancelCtx()
 		runErr := j.Run(jobContext)
-		var updateErr error
```

Comment:
Is there still a possible race condition here since we don't "wait" for the `j.Run(jobContext)`?  Can jobs be writing to the database when the goroutine ends?

Reply:
Not sure I follow?  Run is a blocking call once it returns the job has exited

Reply:
Run is blocking but it's invoked within a goroutine.  I'm just pondering if that coroutine's parent completes without waiting for it, does it have any affect on possible DB writes within the job that's being "Run".

Reply:
Aha I see, yeah it could.  Ill ping you when ready for review.